core/src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -2334,3 +2335,27 @@ strip eq arr@Array{..} = do
             then getLast prev low
             else return cur
         else return cur
+
+-- | Bubble sorts the elements.
+--
+-- /Pre-release/
+{-# INLINE bubbleAsc #-}
+bubbleAsc :: (MonadIO m, Storable a) =>
+    (a -> a -> Ordering) -> Array a -> m ()
+bubbleAsc cmp0 arr = do
+    x <- getIndexUnsafe (l - 1) arr
+    go x (l-2)
+
+    where
+
+    l = length arr
+
+    go x i =
+        when (i >= 0) $ do
+            x1 <- getIndexUnsafe i arr
+            case x1 `cmp0` x of
+                GT -> do
+                    _ <- putIndexUnsafe i x arr
```

Comment:
We can avoid writing this, and write it only in the end when we exit the loop on line 2361.